### ISSUE TITLE
Add support for uncompiled code block arguments and (perhaps) tokens

### ISSUE THREAD

**Issue by qiemem** [NetLogo/NetLogo#862]
The idea here is to accept code blocks as arguments that an extension can then just process as a list of tokens, so that it can be transformed before running (sort of like a macro... that runs at runtime).

If you've got

```
foo:run-on-x-arg [
  ask turtles [ fd x ]
] 5
```

the `foo` extension would receive the list of tokens `[ask, turtles, [, fd, x, ], ]`, which it could the preprocess (in this case, by replace `x` with `5`).

The main user of this will likely be NetLogo/LevelSpace, which needs it in order to pass code blocks between models with different procedures and variables.

Allowing token arguments would let you do similar things with tokens. For instance:

```
(foo:run-with-arg x 5 [
  ask turtles [ fd x ]
])
```

The first argument here would be received as a token. NetLogo/NW-Extension does this in a hacky way with strings (representing link variables), so it would definitely make use of this. Currently, the NetLogo compiler does support this kind of thing with patch variables, which the NetLogo/GIS-Extension uses.

**Comment by qiemem** [comment]
@frankduncan made a first pass at this on the codeblock branch. However, currently that implementation relies on passing `forgiving = true`. This breaks certain expressions containing `+` due to NetLogo 3 backwards compatibility.

Currently, we're discussing removing this backwards compatibility, and removing identifier verification from `IdentifierParser`. Instead, we'll do identifier verification in `ExpressionParser`, where we can correctly skip code blocks (it already blows up correctly here, we just need to fix up error messages).

**Comment by qiemem** [comment]
I've continued the work on wip-codeblock.

**Comment by mrerrormessage** [comment]
This was merged prior to 6.0. Closing.
